scripts/workers/submission_worker.py

```diff
@@ -277,12 +283,61 @@ def run_submission(challenge_id, challenge_phase, submission_id, submission, use
     # call `main` from globals and set `status` to running and hence `started_at`
     submission.status = Submission.RUNNING
     submission.save()
-    with stdout_redirect(stdout) as new_stdout, stderr_redirect(stderr) as new_stderr:      # noqa
-        submission_output = EVALUATION_SCRIPTS[challenge_id].evaluate(annotation_file_path,
-                                                                      user_annotation_file_path,
-                                                                      challenge_phase.codename,)
+    try:
+        successful_submission_flag = True
+        with stdout_redirect(stdout) as new_stdout, stderr_redirect(stderr) as new_stderr:      # noqa
+            submission_output = EVALUATION_SCRIPTS[challenge_id].evaluate(annotation_file_path,
+                                                                          user_annotation_file_path,
+                                                                          challenge_phase.codename,)
+
+        if 'result' in submission_output:
+
+            leaderboard_data_list = []
+            for split_result in submission_output['result']:
+
+                # Check if the dataset_split exists for the codename in the result
+                try:
+                    split_code_name = split_result.items()[0][0]  # get split_code_name that is the key of the result
+                    dataset_split = DatasetSplit.objects.get(codename=split_code_name)
+                except:
+                    stderr.write(traceback.format_exc())
+                    successful_submission_flag = False
+                    break
+
+                # Check if the challenge_phase_split exists for the challenge_phase and dataset_split
+                try:
+                    challenge_phase_split = ChallengePhaseSplit.objects.get(challenge_phase=challenge_phase,
+                                                                            dataset_split=dataset_split)
+                except:
+                    stderr.write(traceback.format_exc())
+                    successful_submission_flag = False
+                    break
+
+                leaderboard_data = LeaderboardData()
+                leaderboard_data.challenge_phase_split = challenge_phase_split
+                leaderboard_data.submission = submission
+                leaderboard_data.leaderboard = challenge_phase_split.leaderboard
+                leaderboard_data.result = split_result[dataset_split.codename]
+
+                leaderboard_data_list.append(leaderboard_data)
+
+            # Check to see if length of leaderboard_data_list matches with length of submission_output['result']
```

Comment:
writing comments is actually good, but I will prefer that you write the reason for the check instead of explaining what the code is doing. Explaining a piece of code by writing a comment states that the code is not readable. Here in your case the code is good enough to read, so it will be great if you can write the reason explaining "WHY".